File: ValidBraces.js
function validBraces(braces){
  
  if (braces.length == 0) {return true;}
  var stack = new Array();
  braces = Array.from(braces);

  for (const c of braces) {
    if (c == '(' || c == '[' || c == '{') { braces.push(c);}
    else if (c == ')' && braces.length != 0 && braces.slice(-1)[0] == '(') { braces.pop(); }
    else if (c == ']' && braces.length != 0 && braces.slice(-1)[0] == '[') { braces.pop(); }
    else if (c == '}' && braces.length != 0 && braces.slice(-1)[0] == '{') { braces.pop(); }
    else { return false;}
  }
  
  return true;
}

File: ValidBraces.cpp
#include <stack>
using namespace std;

bool valid_braces(string braces) 
{
  if (braces.length() == 0) {return true;}
  stack<char> allBraces;
  for (char c : braces) {
    if (c == '(' || c == '[' || c == '{') { 
      allBraces.push(c);
    } // if
    else if (c == ')' && !allBraces.empty() && allBraces.top() == '(') { 
      allBraces.pop();
    } // else if
    else if (c == ']' && !allBraces.empty() && allBraces.top() == '[') { 
      allBraces.pop();
    } // else if
    else if (c == '}' && !allBraces.empty() && allBraces.top() == '{') { 
      allBraces.pop();
    } // else if
    else { return false;} // else
  } // for
  return true;
} // valid_braces